packages/be/crons/application-stats.js

```diff
@@ -0,0 +1,228 @@
+// ////////////////////////////////////////////////////////////// getTotalIssues
+const getTotalIssues = async (repo, dataProgramsToken) => {
+  // console.log('🦞 > 🦀 getting total issues')
+  try {
+    const options = {
+      headers: { Accept: 'application/vnd.github+json', 'X-GitHub-Api-Version': '2022-11-28', Authorization: `Bearer ${dataProgramsToken}` }
+    }
+    const response = await Axios.get(`https://api.github.com/search/issues?q=repo:${repo}%20is:issue`, options)
+    return response.data.total_count
+  } catch (e) {
+    console.log('================================== [Function: getTotalIssues]')
+    console.log(e)
+    throw e
+  }
+}
+
+// ///////////////////////////////////////////////////////////// getApplications
+
+const getApplications = async (applicationType, n = 1, applications = [], totalIssues = 0) => {
+  console.log(`🦞 getting applications: request ${n}`)
+  try {
+    const dataProgramsToken = process.env.GITHUB__PERSONAL_ACCESS_TOKEN__DATA_PROGRAMS
+    const repo = MC.repos[applicationType][0]
+    if (totalIssues === 0) { totalIssues = await getTotalIssues(repo, dataProgramsToken) }
+
+    const options = {
+      headers: { Accept: 'application/vnd.github+json', 'X-GitHub-Api-Version': '2022-11-28', Authorization: `Bearer ${dataProgramsToken}` },
+      params: {
+        state: 'all',
+        per_page: 100,
+        page: n
+      }
+    }
+    const response = await Axios.get(`https://api.github.com/repos/${repo}/issues`, options)
+    const updatedApplications = [...applications, ...response.data]
+    if (updatedApplications.length < totalIssues) {
+      return getApplications(applicationType, n + 1, updatedApplications, totalIssues)
+    }
+    return updatedApplications
+  } catch (e) {
+    console.log('================================= [Function: getApplications]')
+    console.log(e)
+    throw e
+  }
+}
+
+// //////////////////////////////////////////// iterateApplicationLabels
+const iterateApplicationLabels = (application, applicationType) => {
+  // console.log('🦊 > 🐡 determining application state')
+  try {
+    const labels = application.labels
+    const completedRegex = { ga: /(state:)?\s?granted/gi, lda: /total\s?datacap\s?reached/gi }
+    const validatedRegex = { ga: /(bot:)?\s?looking\s?good/gi, lda: /validated/ }
+    const inReviewRegex = { ga: /(state)|(bot)?:\s?(further\s?info)?(review)?\s?needed/gi, lda: /error/ }
+
+    const hasLabel = (labels, regex) => { return labels.some((label) => { return label.name.match(regex) }) }
+
+    const completed = hasLabel(labels, completedRegex[applicationType])
+    const validated = hasLabel(labels, validatedRegex[applicationType])
+    const inReview = hasLabel(labels, inReviewRegex[applicationType])
+    const isPR = application.pull_request
+
+    const applicationState = isPR ? false
+      : completed ? 'completed'
+        : validated ? 'validated'
+          : inReview ? 'inReview'
+            : 'noRelevantLabels'
```

Comment:
Instead of this block, let's import and use your `DetermineApplicationState` helper function:

```js
const { DetermineApplicationState } = require('@Module_Application/logic/determine-application-state')
```

Reply:
I'm still going to need the `hasLabel` function here to check the `source:filplus.storage` and `efil+` labels, should I make another helper function file in `/logic` for it so it's not redefined a bunch of places?

Reply:
@tikagan pulling out `hasLabel` as its own helper function sounds great! :)